pcl_ros/CMakeLists.txt

```diff
@@ -1,245 +1,89 @@
-  src/pcl_ros/filters/project_inliers.cpp
-  src/pcl_ros/filters/radius_outlier_removal.cpp
-  src/pcl_ros/filters/statistical_outlier_removal.cpp
-  src/pcl_ros/filters/voxel_grid.cpp
-  src/pcl_ros/filters/crop_box.cpp
-)
-target_link_libraries(pcl_ros_filters pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_filters ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_filters)
-
-## Declare the pcl_ros_segmentation library
-add_library (pcl_ros_segmentation
-  src/pcl_ros/segmentation/extract_clusters.cpp
-  src/pcl_ros/segmentation/extract_polygonal_prism_data.cpp
-  src/pcl_ros/segmentation/sac_segmentation.cpp
-  src/pcl_ros/segmentation/segment_differences.cpp
-  src/pcl_ros/segmentation/segmentation.cpp
-)
-target_link_libraries(pcl_ros_segmentation pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_segmentation ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_segmentation)
-
-## Declare the pcl_ros_surface library
-add_library (pcl_ros_surface
-  src/pcl_ros/surface/surface.cpp
-  # Compilation is much faster if we include all the following CPP files in surface.cpp
-  src/pcl_ros/surface/convex_hull.cpp
-  src/pcl_ros/surface/moving_least_squares.cpp
-)
-target_link_libraries(pcl_ros_surface ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_surface ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_surface)
-
-## Tools
-
-add_executable(pcd_to_pointcloud tools/pcd_to_pointcloud.cpp)
-target_link_libraries(pcd_to_pointcloud ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(pointcloud_to_pcd tools/pointcloud_to_pcd.cpp)
-target_link_libraries(pointcloud_to_pcd ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(bag_to_pcd tools/bag_to_pcd.cpp)
-target_link_libraries(bag_to_pcd pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(convert_pcd_to_image tools/convert_pcd_to_image.cpp)
-target_link_libraries(convert_pcd_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(convert_pointcloud_to_image tools/convert_pointcloud_to_image.cpp)
-target_link_libraries(convert_pointcloud_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-## Downloads
-
-catkin_download(table_scene_lms400.pcd http://download.ros.org/data/pcl/table_scene_lms400.pcd
-  DESTINATION ${CMAKE_CURRENT_SOURCE_DIR}/samples/data
-  MD5 546b5b4822fb1de21b0cf83d41ad6683
-)
-add_custom_target(download ALL DEPENDS download_extra_data)
+target_include_directories(pcl_ros_tf
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+    ${PCL_INCLUDE_DIRS})
+target_link_libraries(pcl_ros_tf ${PCL_LIBRARIES})
+ament_export_libraries(pcl_ros_tf)
 
 #############
 ## Testing ##
 #############
 
-if(CATKIN_ENABLE_TESTING)
```

Comment:
Are none of these tests enabled by the pcl_ros_tf that you ported?

Reply:
Correct. I started down enabling the tests under `pcl_ros\src\test\test_tf_message_filter_pcl.cpp` but determined that it depends on the pcl_ros_filters module, and decided to take it one step at a time.

However on a second look, it appears I was wrong and it only depends on the *tf2* filters package. It does depend on `pcl_ros\point_cloud.h` though, which has not been ported by this PR.

I'm leaning towards expanding the scope of this PR a bit to get `point_cloud.h` and the first round of unit tests ported as well -- cool? Or would you rather keep the PR smaller?

Reply:
Sounds good - if its just 1 header file extra, that's OK

Reply:
Ah, I've refreshed my memory as to where this started to turn into an onion. The tests make use of `pcl_ros/pointcloud.h` which plugs `pcl::PointCloud<T>` into the message serialization/adaptation feature of ROS1. Seems like this isn't yet implemented in ROS2 and the comments on [this thread](https://answers.ros.org/question/356542/ros2-message-serialization-adapting-types/) suggest it might not be a good idea moving forward anyway.

Thoughts? I could modify the tests to use regular ROS message types, but that would kind of fork away from the current implementation.

Reply:
Turns out I'm just standing on the shoulders of giants --  https://discourse.ros.org/t/ros2-type-masquerading-status-and-alternatives/9461

Reply:
What's the question exactly? I agree that type masquerading isn't available in ROS2 (yet). Does that stop this port or just this file needs to be commented out or something?

Reply:
Sorry my morning was rushed; should have been more verbose.

The good news is, as far as I can tell, the only component _within_ `pcl_ros` itself which actually uses the type-masquerading header is the `pcl_ros\src\test\test_tf_message_filter_pcl.cpp` which is the test file I've proposed enabling in this thread. So the port can go on, but downstream consumers of pcl_ros which rely on the type masquerading stuff obviously won't work. I don't have any context to tell how critical of a feature this is in the `pcl_ros` library.

As I see it, the options are:

1. Skip porting the test for now until the masquerading is available
2. Port the test now with a bit of rewriting to use `sensor_msgs::PointCloud2` messages instead of `pcl::PointCloud<T>` directly (and down the road change it back to get coverage on the masquerading)
3. Go start shaving the type masquerading yak now

My vote is #2 to get some automated coverage and keep the `pcl_ros` port on track. Though type masquerading would have widespread benefit it seems.

Reply:
Could you just change the type from the pcl::Pointcloud to the sensor_msgs::Pointcloud2 to avoid type masquerading? Obviously the test will be slower, but that's OK given the state of things. 

Its pretty important for performance there's no conversions from PC2 -> pcl -> PC2 to publish again, but for now we can only do what we can do. PCL now has this `PCLPointCloud2` thing which my understanding is doesn't have to copy around and instead just reformats the data for a limited subset of the filters but works with most here (or slightly more efficient).

Reply:
We can also file a ticket to re-add type masquerading when available, but that might be a ways down the road from the last few discussions I've had with open robotics about this (its hard, time consuming, and so far no one funding it).

Reply:
So after starting down this road, I realized these tests aren't actually testing anything other than the type masquerading itself. It's using the `tf2` library as a vehicle, but it is NOT actually testing the `pcl_ros` transforms library.

With that in mind I've held off on porting these tests.

Unfortunately there are no unit tests specifically for the transforms library, but they're sufficiently simple wrappers around PCL and I've gone through and manually tested them. We can file a ticket to add unit testing as an effort outside of the porting if needed.